### PR TITLE
feature_release states:167 add states to en/disable xdebug; also add a flag '-x' to install xdebug on instance creation

### PR DESCRIPTION
**Pull request by daniwe4**
https://github.com/conceptsandtraining/doil/issues/167

### CODE REVIEW

File: setup/stack/states/enable-xdebug/xdebug/init.sls [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/etc/php/{{ maj }}.{{ min }}/apache2/php.ini:
+  file.append:
+    - text:
+      - ; XDEBUG-START
+      - xdebug.mode = develop,debug,profile
+      - xdebug.discover_client_host = true
+      - xdebug.client_port = 9000
+      - xdebug.log = /var/log/xdebug.log
+      - xdebug.start_with_request = trigger
+      - xdebug.output_dir = /var/log/xprofiles"
```
Comment (tbongers-cat) [inline setup/stack/states/enable-xdebug/xdebug/init.sls:18]: There are leftover quotation marks at the end of the line.

File: README.md [hunk 2]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
...
+
+#### Necessary PHP-Storm plugins:
+
+* PHP Docker
+* PHP Remote
+
+#### PHP-Storm settings
+
+* it is best to open the doil instance root directory in PHP-Storm.
+* under Settings PHP CLI interpreter: add a new interpreter (on the 3 dots and then in the pop up on +)
```
Comment (tbongers-cat) [inline README.md:339]: "Settings > PHP > CLI Interpreter" (makes it a bit more readable)

File: README.md [hunk 3]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
...
+* select "From Docker, Vagrant..." and then select "docker-compose".
```
Comment (tbongers-cat) [inline README.md:340]: The option is called "Docker Compose"

File: README.md [hunk 4]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
...
+* if you already have the doil instance root open in phpstorm, it automatically selects the docker-compose.yml
```
Comment (tbongers-cat) [inline README.md:341]: We should call it "PhpStorm" everywhere to be consistent

File: README.md [hunk 5]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
...
+* select the doil name under Service (e.g. ilias7) and click "OK".
```
Comment (tbongers-cat) [inline README.md:342]: "doil instance name"

File: README.md [hunk 6]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
...
+* click on the reload icon under General PHP executable and the warnings should disappear. Then with "OK" the pop-ups again
```
Comment (tbongers-cat) [inline README.md:343]: "Then click 'OK' to close the pop-ups."

File: README.md [hunk 7]
```diff
@@ -303,6 +303,56 @@ users, so make sure to understand what you are doing.
 * `doil mail:stop` stops the mail server
 * `doil mail:restart` restarts the mail server
 
+### xdedug
+
+**doil** provides two options to enable xdebug for the given instance.  
+
+First an instance can be created with the '-x' flag, or alternatively
```
Comment (tbongers-cat) [inline README.md:310]: "Option 1: An instance can ..."
[...]
"Option 2: You can apply ..."

File: setup/stack/states/enable-xdebug/xdebug/init.sls (hunk 1 above)
Comment (daniwe4) [reply setup/stack/states/enable-xdebug/xdebug/init.sls:18]: Done!

File: README.md (hunk 2 above)
Comment (daniwe4) [reply README.md:339]: Done!

File: README.md (hunk 3 above)
Comment (daniwe4) [reply README.md:340]: Done!

File: README.md (hunk 4 above)
Comment (daniwe4) [reply README.md:341]: Done!

File: README.md (hunk 5 above)
Comment (daniwe4) [reply README.md:342]: Done!

File: README.md (hunk 7 above)
Comment (daniwe4) [reply README.md:310]: Done!

**Review by tbongers-cat — APPROVED** [review]

**Pull request merged**